cscs-checks/apps/greasy/greasy_check.py

```diff
@@ -0,0 +1,218 @@
+# Copyright 2016-2020 Swiss National Supercomputing Centre (CSCS/ETH Zurich)
+# ReFrame Project Developers. See the top-level LICENSE file for details.
+#
+# SPDX-License-Identifier: BSD-3-Clause
+
+import itertools
+import os
+import fnmatch
+
+from datetime import datetime
+
+import reframe as rfm
+import reframe.utility.sanity as sn
+
+from reframe.core.launchers.registry import getlauncher
+
+
+def toSeconds(str):
+    return (datetime.strptime(str, '%H:%M:%S') -
+            datetime.strptime('00:00:00', '%H:%M:%S')).total_seconds()
+
+
+@rfm.required_version('>=2.19')
+@rfm.parameterized_test(
+    ['serial',     'daint:gpu', 24, 12, 1, 1],
+    ['serial',     'daint:mc',  72, 36, 1, 1],
+    ['openmp',     'daint:gpu', 24,  3, 1, 4],
+    ['openmp',     'daint:mc',  72,  9, 1, 4],
+    ['mpi',        'daint:gpu', 24,  4, 3, 1],
+    ['mpi',        'daint:mc',  72, 12, 3, 1],
+    ['mpi+openmp', 'daint:gpu', 24,  3, 2, 2],
+    ['mpi+openmp', 'daint:mc',  72,  6, 3, 2])
+class GREASYCheck(rfm.RegressionTest):
+    def __init__(self, variant, system, num_greasy_tasks, nworkes_per_node,
+                 nranks_per_worker, ncpus_per_worker):
+
+        self.valid_systems = [system]
+        if system.startswith('daint'):
+            self.valid_systems += [system.replace('daint', 'dom')]
+
+        self.valid_prog_environs = ['PrgEnv-gnu']
```

Comment:
Shouldn't that be `builtin`?

Reply:
@victorusu What about this?

Reply:
It has to be `PrgEnv-gnu` because we are compiling the `hello, world` applications. And there is a variant with MPI.